cmd/booster-http/gateway_handler.go

```diff
@@ -52,34 +58,61 @@ func (h *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if h.nitroRpcClient != nil {
+		// This the payment we expect to receive for the file.
+		const expectedPayment = int64(5)
 
 		params, _ := url.ParseQuery(r.URL.RawQuery)
-		if !params.Has("channelId") {
-			webError(w, fmt.Errorf("a valid channel id must be provided"), http.StatusPaymentRequired)
+
+		v, err := parseVoucher(params)
+		if err != nil {
+			webError(w, fmt.Errorf("could not parse voucher: %w", err), http.StatusBadRequest)
 			return
 		}
-		rawChId := params.Get("channelId")
-
-		chId := types.Destination(common.HexToHash(rawChId))
 
-		// TODO: Allow this to be configurable
-		expectedPaymentAmount := big.NewInt(10)
+		// delta is the change in the channel balance caused by adding this voucher.
+		_, delta := h.nitroRpcClient.ReceiveVoucher(v)
 
-		hasPaid, err := checkPaymentChannelBalance(h.nitroRpcClient, chId, expectedPaymentAmount)
-		if err != nil {
-			webError(w, err, http.StatusPaymentRequired)
+		// TODO: A nil value indicates an error with the voucher. We should update to the latest go-nitro which properly returns the error.
```

Comment:
Should we do this now or in future work?

Reply:
I think we can punt this out of the PR but should do this soon. Created https://github.com/statechannels/boost/issues/18